File: gaia-sdk-python/graphql/request/input/PerceiveStreamImpulse.py


from typing import Callable
from api.VariableRegistry import VariableRegistry


from dataclasses import dataclass


@dataclass
class PerceiveStreamImpulse():
    """
    Input for stream perception impulse
    """

    broadcast: Boolean
    identityId: Uuid
    type: String
    data: Struct

File: gaia-sdk-javascript/src/graphql/request/input/PerceiveStreamImpulse.ts
import {Struct, Uuid} from "../../GaiaClient";

/**
 * Input for stream perception impulse
 */
export default class PerceiveStreamImpulse {

    private broadcast:Boolean;
    private identityId:Uuid;
    private type:String;
    private data:Struct;

    constructor (broadcast:Boolean, identityId:Uuid, type:String, data:Struct) {
        this.broadcast = broadcast;
        this.identityId = identityId;
        this.type = type;
        this.data = data;
    }
}
